docs/en/concepts-and-designs/mal.md

```diff
@@ -0,0 +1,188 @@
+# Meter Analysis Language
+
+Meter system provides a functional analysis language called MAL(Meter Analysis Language) that lets the user analyze and 
+aggregate meter metric data in OAP streaming system. The result of an expression can either be ingested by agent analyzer,
+or OC/Prometheus analyzer.
+
+## Language data type
+
+In MAL, an expression or sub-expression can evaluate to one of two types:
+
+ - Sample family -  a set of samples(metrics) containing a range of metrics whose name is identical.
+ - Scalar - a simple numeric value. it supports integer/long, floating/double,
+
+## Sample family
+
+A set of samples, which is as the basic unit in MAL. For example:
+
+```
+instance_trace_count
+```
+
+The above sample family might contains following simples which are provided by external modules, for instance, agent analyzer:
```

Comment:
simples? A typo?

Reply:
Where is the term `sample family` from?

Reply:
From OC and Prometheus term system, `sample family` contains a series of sample/metric with the same name and tag keys. Those samples in an identical `sample family` could be aggregated using `aggregation operation` by tags.

In OC, `sample family` is called `metric family`, but it also uses `sample` to refer to `metric`. So I use the same term, `sample`, to identify them instead of introducing the `metric` keyword.